src/shared-components/alert/index.tsx

```diff
@@ -78,36 +96,34 @@ class Alert extends React.Component {
       return;
     }
 
-    // eslint-disable-next-line no-undef
     this.timer = window.setTimeout(
       this.alertExitHandler,
       Number(duration) * 1000 - ANIMATION_DELAY,
     );
   }
 
-  componentWillUnmount() {
+  componentWillUnmount(): void {
     if (this.timer) {
-      // eslint-disable-next-line no-undef
       window.clearTimeout(this.timer);
     }
   }
 
-  alertExitHandler = () => {
+  alertExitHandler = (): void => {
     const { onExit, ...rest } = this.props;
     this.setState({ exiting: true });
 
-    // eslint-disable-next-line no-undef
     window.clearTimeout(this.timer);
 
-    // eslint-disable-next-line no-undef
     window.setTimeout(() => {
       this.setState({ exited: true });
 
-      onExit({ ...rest });
+      if (onExit) {
```

Comment:
If we're doing a truthy check for `onExit` now, due to typing, I think I'd prefer we type it in `defaultProps` as `undefined`. (This is consistent with our TypeScript type of `onExit?` as well.) I didn't see any typing issues in my terminal when doing that.